src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgFulfilmentRequestProcessor.java

```diff
@@ -0,0 +1,133 @@
+package uk.gov.ons.census.fwmt.outcomeservice.converter.spg;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+import uk.gov.ons.census.fwmt.outcomeservice.config.GatewayOutcomeQueueConfig;
+import uk.gov.ons.census.fwmt.outcomeservice.converter.SpgOutcomeServiceProcessor;
+import uk.gov.ons.census.fwmt.outcomeservice.data.GatewayCache;
+import uk.gov.ons.census.fwmt.outcomeservice.data.GatewayCache.GatewayCacheBuilder;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.FulfilmentRequestDto;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.SpgOutcomeSuperSetDto;
+import uk.gov.ons.census.fwmt.outcomeservice.message.GatewayOutcomeProducer;
+import uk.gov.ons.census.fwmt.outcomeservice.service.impl.GatewayCacheService;
+import uk.gov.ons.census.fwmt.outcomeservice.template.TemplateCreator;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.integration.common.product.ProductReference;
+import uk.gov.ons.ctp.integration.common.product.model.Product;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_ADDRESS_NOT_VALID_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.EventType.FULFILMENT_REQUESTED;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.SurveyType.spg;
+import static uk.gov.ons.ctp.integration.common.product.model.Product.RequestChannel.FIELD;
+
+@Slf4j
+@Component("FULFILMENT_REQUESTED")
+public class SpgFulfilmentRequestProcessor implements SpgOutcomeServiceProcessor {
+
+  @Autowired
+  private ProductReference productReference;
+
+  @Autowired
+  private GatewayOutcomeProducer gatewayOutcomeProducer;
+
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  @Autowired
+  private GatewayCacheService gatewayCacheService;
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    UUID caseId = (outcome.getCaseId() != null) ? outcome.getCaseId() : caseIdHolder;
+    if (outcome.getFulfillmentRequests() == null) {
+      return caseIdHolder;
+    }
+    for (FulfilmentRequestDto fulfilmentRequest : outcome.getFulfillmentRequests()) {
+      if (!isQuestionnaireLinked(fulfilmentRequest)) {
+
+        String eventDateTime = outcome.getEventDate().toString();
```

Comment:
Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

Reply:
Done.